### PR TITLE
Support for mix of absolute and relative paths when generating sourcemaps

### PR DESCRIPTION
**Pull request by jlmatchlock**
See https://github.com/kripken/emscripten/issues/2970#issuecomment-311937107

### CODE REVIEW

File: lib/Target/JSBackend/JSBackend.cpp [hunk 1]
```diff
@@ -728,8 +729,16 @@ namespace {
         auto *Scope = cast_or_null<DIScope>(Loc.getScope());
         if (Scope) {
           StringRef File = Scope->getFilename();
-          if (Line > 0)
-            Code << " //@line " << utostr(Line) << " \"" << (File.size() > 0 ? File.str() : "?") << "\"";
+		  if (Line > 0)
+		  {
```
Comment (kripken) [inline lib/Target/JSBackend/JSBackend.cpp]: `{` should be on preceding line (see other code around here)

also, looks like you are using tabs perhaps? indentation should be 2 spaces.

File: lib/Target/JSBackend/JSBackend.cpp [hunk 2]
```diff
@@ -728,8 +729,16 @@ namespace {
...
+			  const StringRef Dir = Loc->getDirectory();
+			  const Twine currentFilePath(File);
+
+			  if (llvm::sys::path::is_absolute(currentFilePath))
```
Comment (kripken) [inline lib/Target/JSBackend/JSBackend.cpp]: please use `{`, `}` around the if and else bodies

**Comment by jlmatchlock** [comment]
Just to let you know, been busy with work but I'll get around this as soon as I can.

**Comment by vvuk** [comment]
Happy to help get this moving again if needed -- @jlmatchlock let me know!

**Comment by jlmatchlock** [comment]
Sorry for the very long delay. I'm still busy with other tasks and haven't gotten around this yet.

I believe the only one remaining task is to write an unit test for this.
I'll need to confirm later but off the top of my head, the environment to reproduce the issue looked like this:
- a project with two libraries and one application (although one library should be enough)
- libraries have relative paths from the building directory to their cpp and include files
- application source files has absolute paths

I'll try to reproduce the issue with my current setup (which has changed a bit since 6 months ago) but if you could give some pointers regarding how to setup the test case this could expedite the merge (I don't have much experience in python)

Thank you for the reminder.

**Pull request closed without merging**